### ISSUE TITLE
Problem with resuming DSMC case

### ISSUE THREAD

**Issue by jaimakhijaa** [vincentcasseau/hyStrath#73]
Hello,
I have a DSMC case which I am unable to resume running the case after I stop it once. The case was set up without any errors, right from meshing, initializing, decomposing to run the case in parallel, renumbering the mesh, and run the case itself using mpirun. The initial run was for a flow time of 0.28 seconds, where I stopped the case by modifying the controlDict file by changing the endTime to writeNow. When I resumed the case by calling the mpirun command again (not modifying anything within the processor folders), I faced an error of floating-point exception. This is an inconvenience as restarting the simulation from time = 0 renders the initial run data useless. I have also attached a log file showing the error. Any help with this would be really helpful.

Regards,
Jai
[log.txt](https://github.com/vincentcasseau/hyStrath/files/6365998/log.txt)

**Comment by timteichmann** [comment]
Hello Jai,
One possible reason I can think of is that the `controlDict` is also present in the decomposed processor folder. Can you check if this is the case (if yes there will be one or more files at `processorN/system/controlDict`).
However, from personal experience I can recommend to set up the simulation once and then let it run without changing any parameters (except the starting of the averaging ofc). Many of the settings in the original OpenFOAM are not respected by dsmcFoam+ and some of them might lead to crashes like the one you described.

From your log I can tell that it is probably failing in

`hyStrath/src/lagrangian/molecularDynamics/general/time/timeDataMeas/timeDataMeas.C` line 361 ff.

Maybe this helps you locate the exact cause of this bug.

**Comment by jaimakhijaa** [comment]
Hello Tim,
Thank you for responding to my query. I did have a look in multiple processor folders and other than the time steps, the folders present are 'constant', 'boundaries', and 'fieldMeasurements'. So I don't think the controlDict file is present there and therefore, not the cause of the error.
I shall look into the file whose directory you have shared. Could you tell me what you saw in the log file that you suggested I have a look in the _timeDataMeas.C_ file? Neither the file name nor the directory was mentioned in the log file.

Regards,
Jai

**Comment by vincentcasseau** [comment]
Hi Jai,
Could you please provide a minimal, reproducible test case?
Regards,
Vincent

**Comment by vincentcasseau** [comment]
Here's a summary of where we are:
- [x] removing regression cases and leaving only 5 test cases (axisymmetric blunt cylinder, orion capsule, couette flow, hypersonic corner, supersonic flat plate), as discussed in #71;
- [x] updating test cases: sampling at time: #55, load balancing until time: #56 (not used, but entry added in _loadBalancingDict_)
- [ ] fix the bug in the calculation of the mixture vibrational temperature
- [ ] fix the dsmcN_* and dsmcNMean_* fields at boundaries
- [ ] provide results folder for the orion and couette test cases
- [ ] fix the bug that prevents from resuming DSMC simulations (_averagingAcrossManyRuns_ is currently disabled)
- [ ] test mesh adaptation on the orion test case

**Comment by jaimakhijaa** [comment]
Yeah sure, have zipped it in here. I have removed few files to upload the files within the limit of GitHub (10 MB). Also, I would like to know if there are any differences between how rhoNMean is calculated in dsmcFoam and dsmcFoam+. I cannot calculate macroscopic variables such as velocity, temperature, and pressure in dsmcFoam but dsmcFoam+ can calculate the same.

Regards,
Jai

[hollow_cylinder_flare_dsmcFoam+.zip](https://github.com/vincentcasseau/hyStrath/files/6470951/hollow_cylinder_flare_dsmcFoam%2B.zip)

**Comment by timteichmann** [comment]
Thanks for the fixes Vincent! Also apologies that my changes did apparently create issues with the hybrid code after all.
Do you think that the bug regarding `averagingAcrossManyRuns` is related with #55 or # 56?

**Comment by vincentcasseau** [comment]
Not at all Tim. It was an excellent step in the right direction. This file, and many others that you flagged as deprecated (WIP), are/were chaotic.
`averagingAcrossManyRuns` actually works in serial. I compared the current implementation to the one before #55 and I can't see a problem. It makes me wonder if it has ever been tested in parallel...
I will check out dsmcFoam+ as of last September and test it later this week.

**Comment by vincentcasseau** [comment]
In the next commit, I am planning on renaming a few fields that are unnecessarily long:
- translationalT -> Ttra
- translationalT -> Ttra
- translationalT -> Ttra
- translationalT -> Ttra

**Comment by vincentcasseau** [comment]
Hi Jai,
This is fixed. Please have a look at this comment https://github.com/vincentcasseau/hyStrath/issues/73#issuecomment-840176373
I'll close this issue once all checkboxes are ticked.
Thanks,
Vincent

**Comment by JaiMakhija73** [comment]
Thank you for your help Vincent and Tim! Also could you also share the paper from which you have validated the Orion CEV case and built the tutorial? I have not founded the paper on your website.

**Comment by jaimakhijaa** [comment]
Thank you for your help Vincent and Tim!
Could you share the paper which you've used for validating the Orion CEV case and building the tutorial? I couldn't find it in your website.

**Comment by vincentcasseau** [comment]
On the Orion capsule, we have @rophys's PhD thesis:
R. C. Palharini, 09/2014: [Atmospheric Reentry Modelling Using an Open-Source DSMC Code](https://github.com/vincentcasseau/hyStrath/blob/master/doc/PhDthesis-rodrigopalharini.pdf), pp. 86-99
but the altitude and the angle of attack look different.
Do you have a conf. paper or journal article on the Orion, Rodrigo?

**Comment by rophys** [comment]
Hi Vicent,

The Orion case is based on a paper published by Moss in 2006. Please, find
it attached.

Cheers,

Rodrigo

**Comment by jaimakhijaa** [comment]
Hi @rophys, 
Is this the paper you're talking about? [https://arc.aiaa.org/doi/10.2514/6.2006-8081](url)
Can't find any attachment in your comment.

**Comment by vincentcasseau** [comment]
It looks like it:
R. C. Palharini, 09/2014: [Atmospheric Reentry Modelling Using an Open-Source DSMC Code](https://github.com/vincentcasseau/hyStrath/blob/master/doc/PhDthesis-rodrigopalharini.pdf), Ref 193
[193] J. N. Moss, K. A. Boyles, and F. A. Greene. Orion aerodynamics for hypersonic free
molecular to continuum conditions. In 14th AIAA/AHI International Space Planes and
Hypersonic Systems and Technologies Conference. Canberra, Australia, 2006.

The attachment didn't work because he replied by email. However, please do not attach papers here for copyright reasons.

**Comment by rophys** [comment]
Hi Jai,

Yes, it is the correct paper. You will find all the information that you
need in it. In addition, you can have a look at my PhD thesis
https://pureportal.strath.ac.uk/en/publications/atmospheric-reentry-modelling-using-an-open-source-dsmc-code
.

**Comment by vincentcasseau** [comment]
I will open a new issue and close this one for clarity.

**Comment by jaimakhijaa** [comment]
Hello @vincentcasseau 
I am facing an issue in starting a new simulation now using dsmcFoam+. I have attached the log file for your reference. The case setup was modified to run the case using staticFvMesh and the error of floating point exception was generated. Any form of help would be appreciated.

Regards,
Jai
[log.txt](https://github.com/vincentcasseau/hyStrath/files/6648641/log.txt)

**Comment by vincentcasseau** [comment]
Hi Jai,
> The case setup was modified to run the case using staticFvMesh

Why would you like to do this?

**Comment by jaimakhijaa** [comment]
> Hi Jai,
> 
> Why would you like to do this?

I did it for faster simulation time as this is a trial run, not the final run. I also did try running with the default dynamicMeshDict but did face the same floating point exception error.

**Comment by vincentcasseau** [comment]
Are the tutorial cases running? i.e., is your code correctly installed?
There are no useful info in the log.

**Comment by jaimakhijaa** [comment]
I just tried running the orion107kmNR tutorial and did get an error in running the case using the Allrun script. Didn't face any error during installation though. Any idea what might have caused it? Attached the log for the tutorial case in case it helps.
[log_dsmcFoam+.txt](https://github.com/vincentcasseau/hyStrath/files/6649341/log_dsmcFoam%2B.txt)

**Comment by vincentcasseau** [comment]
Change dir to _username-v1706/src/_ and  _username-v1706/applications/_ and type in both
`wclean all`
Then, sync the DSMC module in hyStrath/ using the install script.

**Comment by jaimakhijaa** [comment]
I just tried what you've mentioned above and I'm getting the same error for the orion107kmNR tutorial. Didn't face any error during synchronization though. Any idea what might have caused it? Attached the log for the tutorial case in case it helps.
[log_orion107kmNR.txt](https://github.com/vincentcasseau/hyStrath/files/6720051/log_orion107kmNR.txt)

**Comment by vincentcasseau** [comment]
Hi,
I reinstalled everything from scratch at my end and it works. As anticipated, there are no issues with the current commit.
So you should do the same:
1. delete src/, applications/ and platforms/ in $WM_PROJECT_USER_DIR;
2. run git pull
3. install what you need
Thanks,
Vincent
